### DISCUSSION TITLE
possibility of supporting git-remote-gcrypt?

### DISCUSSION THREAD

**Discussion by tejing1** [zeapo/Android-Password-Store#1443] [Ideas]
I just started using APS, and was generally quite pleased with its feature set. The one complication for me was that I couldn't use my gcrypted repository on rsync.net as the origin to sync with, and instead had to set up dynamic dns and a port forward so I could sync with my home desktop from the outside. (realistically I was going to set up external ssh access anyway, so it's no great bother)

I don't particularly like leaking even which sites I have accounts on outside of my actual devices, so I use git-remote-gcrypt for my remote password repo so that even the folder structure, filenames, and commit history are obscured. This seems a very relevant capability for the password store use-case, and shouldn't require any major new components, since you've already got a key management and encryption backend in place, so I was wondering if there's any interest in adding support for it in APS?

**Comment by msfjarvis** [comment]
We can't support it because the Git library we currently use does not support git remote helpers, and it's quite unlikely that we'll ever be able to remedy that situation. This is the same limitation that prevents support for [Keybase](https://github.com/android-password-store/Android-Password-Store/issues/344).

**Reply by tejing1** [reply]
Ah, I see. Harder than it at first seemed. Oh well.

**Reply by bjmgeek** [reply]
What git library are you using?

**Reply by msfjarvis** [reply]
[JGit](https://www.eclipse.org/jgit/) v3.7.1.201504261725-r
